lib/tcpip/BufferManagement.cc

```diff
@@ -49,16 +66,19 @@ pxGetNetworkBufferWithDescriptor(size_t     xRequestedSizeBytes,
 
 	Timeout t{xBlockTimeTicks};
 
-	std::unique_ptr<NetworkBufferDescriptor_t> descriptor{
+	// TODO we likely want to pre-allocate (or re-use) the descriptors at
+	// some point
+	auto deleter = [=](void *ptr) { heap_free(BM_MALLOC, ptr); };
```

Comment:
Can you check that this is hitting the special cases in unique ptr for deletes that are pure functions (and so don’t need to be allocated)?

Reply:
Interesting. I am investigating that.

Reply:
I have looked at different approaches to deleters (default deleter, lambda, std::function, deleter class), and curiously they all seem to have the same memory footprint (8 Bytes on top of the allocation itself).

This seems to contradict some of the stuff I've found online (https://alexanderhoughton.co.uk/blog/c-unique-shared-ptr-custom-deleters/), but we have a different compiler and it's been a few years since the article was published.